### PR TITLE
IGNITE-9732: [Spark] Add joins to Spark Dataframe examples

### PR DESCRIPTION
**Pull request by zaleslaw**

### CODE REVIEW

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+import java.lang.{Integer => JInt, Long => JLong, String => JString}
+
+import org.apache.ignite.cache.query.SqlFieldsQuery
+import org.apache.ignite.configuration.CacheConfiguration
+import org.apache.ignite.spark.IgniteDataFrameSettings._
+import org.apache.ignite.{Ignite, Ignition}
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.sql.SparkSession
+
+/**
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Javadoc not formatted properly.

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  * Example application demonstrates the join operations between two dataframes or Spark tables with data saved in Ignite caches.
+  */
+object IgniteDataFrameJoinExample extends App {
+    /**
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Javadoc not formatted properly. Let's use single line javadoc here

`/** Ignite config file. */`

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+      * Ignite config file.
+      */
+    private val CONFIG = "examples/config/example-ignite.xml"
+
+    /**
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Javadoc not formatted properly. Let's use single line javadoc here

`/** Test cache name. */`

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      * Test cache name.
+      */
+    private val CACHE_NAME = "testCache"
+
+    //Starting Ignite server node.
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: whitespace missed. `// Starting Ignite server node.`

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    val ignite = setupServerAndData
+
+    closeAfter(ignite) { ignite ⇒
+        //Creating spark session.
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: whitespace missed

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+            .master("local")
+            .config("spark.executor.instances", "2")
+            .getOrCreate()
+
+        // Adjust the logger to exclude the logs of no interest.
+        Logger.getRootLogger.setLevel(Level.ERROR)
+        Logger.getLogger("org.apache.ignite").setLevel(Level.INFO)
+
+        // Executing examples.
+
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Empty line not required.

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+        sparkDSLJoinExample
+        nativeSparkSqlJoinExample
+    }
+
+    /**
+      * Examples of usage Ignite DataFrame implementation.
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Javadoc not formatted properly.

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+        val joinResult = persons.join(cities, persons("city_id") === cities("id"))
+          .select(persons("name").as("person"), persons("age"), cities("name").as("city"), cities("country"))
+
+        joinResult.explain(true)
+        joinResult.printSchema()
+        joinResult.show()
+    }
+
+    /**
+      * Examples of usage Ignite DataFrame implementation.
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Javadoc not formatted properly.

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+        // Selecting data from Ignite throw Spark SQL Engine.
+        val joinResult = spark.sql("SELECT person.name AS person, age, city.name AS city, country FROM person JOIN city ON person.city_id = city.id")
+
+        joinResult.explain(true)
+        joinResult.printSchema()
+        joinResult.show()
+    }
+
+    def setupServerAndData: Ignite = {
+        //Starting Ignite.
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Whitespace missed.

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+        val ignite = Ignition.start(CONFIG)
+
+        //Creating first test cache.
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Whitespace missed.

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+        val ccfg = new CacheConfiguration[JLong, JString](CACHE_NAME).setSqlSchema("PUBLIC")
+
+        val cache = ignite.getOrCreateCache(ccfg)
+
+        //Creating SQL tables.
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Whitespace missed.

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+        cache.query(new SqlFieldsQuery(
+            "CREATE TABLE city (id LONG PRIMARY KEY, name VARCHAR, country VARCHAR) WITH \"template=replicated\"")).getAll
+
+        cache.query(new SqlFieldsQuery(
+            "CREATE TABLE person (id LONG, name VARCHAR, age INT, city_id LONG, PRIMARY KEY (id, city_id)) " +
+                "WITH \"backups=1, affinityKey=city_id\"")).getAll
+
+        cache.query(new SqlFieldsQuery("CREATE INDEX on Person (city_id)")).getAll
+
+        //Inserting some data to tables.
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Whitespace missed.

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+
+        persons.printSchema()
+        persons.show()
+
+        // Registering DataFrame as Spark view.
+        persons.createOrReplaceTempView("person")
+        cities.createOrReplaceTempView("city")
+
+        // Selecting data from Ignite throw Spark SQL Engine.
+        val joinResult = spark.sql("SELECT person.name AS person, age, city.name AS city, country FROM person JOIN city ON person.city_id = city.id")
```
Comment (nizhikov) [inline examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Can we use multiline string literal here to make SQL query more readabale?

```
val joinResult = spark.sql(""" 
    | SELECT 
    |   person.name AS person,
    |   age,
    |   city.name AS city,
    |   country
    | FROM
    |   person JOIN 
    |   city ON person.city_id = city.id
""");
```

**Comment by nizhikov** [comment]
@zaleslaw Thanks for the PR.

Almost LGTM.
Please, resolve minor comments according to the code formatting.

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala (hunk 1 above)
Comment (zaleslaw) [reply examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Looks like the comment for a class could be in this form
https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-JavadocComments

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala (hunk 2 above)
Comment (zaleslaw) [reply examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Ok, I will use, but another examples formatted this way (I've copied formatting)

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala (hunk 8 above)
Comment (zaleslaw) [reply examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Is Ignite JavaDoc formatting applied to Scala code?

File: examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala (hunk 13 above)
Comment (zaleslaw) [reply examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala]: Cool advise, thanks

**Pull request merged**